### ISSUE TITLE
Text classification -  error when fitting HyperoptEstimator

### ISSUE THREAD

**Issue by rdoume** [hyperopt/hyperopt-sklearn#71]
Hi there,

I am encountering an issue regarding the text classifier example explained at http://hyperopt.github.io/hyperopt-sklearn/.

I'm litteraly copy-pasting the example, but an error rises when I execute the fit command.

### 
```
estim.fit( X_train, y_train )

Traceback (most recent call last):
  File "<ipython-input-12-8ecb54c50078>", line 1, in <module>
    estim.fit( X_train, y_train )

  File "/home/robin/Documents/Python/hyperopt-sklearn/hpsklearn/estimator.py", line 708, in fit
    fit_iter.send(increment)
  File "/home/robin/Documents/Python/hyperopt-sklearn/hpsklearn/estimator.py", line 619, in fit_iter
    return_argmin=False, # -- in case no success so far
  File "/home/robin/anaconda3/lib/python3.5/site-packages/hyperopt/fmin.py", line 307, in fmin
    return_argmin=return_argmin,
  File "/home/robin/anaconda3/lib/python3.5/site-packages/hyperopt/base.py", line 635, in fmin
    return_argmin=return_argmin)
  File "/home/robin/anaconda3/lib/python3.5/site-packages/hyperopt/fmin.py", line 320, in fmin
    rval.exhaust()
  File "/home/robin/anaconda3/lib/python3.5/site-packages/hyperopt/fmin.py", line 199, in exhaust
    self.run(self.max_evals - n_done, block_until_done=self.async)
  File "/home/robin/anaconda3/lib/python3.5/site-packages/hyperopt/fmin.py", line 173, in run
    self.serial_evaluate()
  File "/home/robin/anaconda3/lib/python3.5/site-packages/hyperopt/fmin.py", line 92, in serial_evaluate
    result = self.domain.evaluate(spec, ctrl)
  File "/home/robin/anaconda3/lib/python3.5/site-packages/hyperopt/base.py", line 840, in evaluate
    rval = self.fn(pyll_rval)
  File "/home/robin/Documents/Python/hyperopt-sklearn/hpsklearn/estimator.py", line 582, in fn_with_timeout
    raise fn_rval[1]

ValueError: zero-dimensional arrays cannot be concatenated 
```
I have checked that my numpy version is 1.11.0 ( I saw that this could be an issue).

Do you have an idea of what's going on? Maybe Python 3.5 ?
Thank you,

**Comment by bjkomer** [comment]
It looks like this bug was introduced during refactoring when a bunch of new features were added. The problem stems from `np.concatenate()` being used on a sparse array. I'll look into it more and see if I can get a fix.

In the meantime, one option would be to use a version of the code before the bug was introduced. [This](https://github.com/hyperopt/hyperopt-sklearn/commit/3393a2236ee1a1ef46a5e474f85532357597aed3) is the last commit where things are working. You can check this point out and use it with this command:
`git checkout 3393a2236ee1a1ef46a5e474f85532357597aed3`

**Comment by rdoume** [comment]
Ok, This partially solve the problem. 
However, now I got this error 
``
Traceback (most recent call last):

  File "<ipython-input-6-5d27e4ef0ae0>", line 6, in <module>
    estim.fit( X_train, y_train )

  File "/home/robin/Documents/Python/hyperopt-sklearn/hpsklearn/estimator.py", line 384, in fit
    fit_iter.send(increment)

  File "/home/robin/Documents/Python/hyperopt-sklearn/hpsklearn/estimator.py", line 353, in fit_iter
    return_argmin=False, # -- in case no success so far

  File "/home/robin/anaconda3/lib/python3.5/site-packages/hyperopt/fmin.py", line 307, in fmin
    return_argmin=return_argmin,

  File "/home/robin/anaconda3/lib/python3.5/site-packages/hyperopt/base.py", line 635, in fmin
    return_argmin=return_argmin)

  File "/home/robin/anaconda3/lib/python3.5/site-packages/hyperopt/fmin.py", line 320, in fmin
    rval.exhaust()

  File "/home/robin/anaconda3/lib/python3.5/site-packages/hyperopt/fmin.py", line 199, in exhaust
    self.run(self.max_evals - n_done, block_until_done=self.async)

  File "/home/robin/anaconda3/lib/python3.5/site-packages/hyperopt/fmin.py", line 173, in run
    self.serial_evaluate()

  File "/home/robin/anaconda3/lib/python3.5/site-packages/hyperopt/fmin.py", line 92, in serial_evaluate
    result = self.domain.evaluate(spec, ctrl)

  File "/home/robin/anaconda3/lib/python3.5/site-packages/hyperopt/base.py", line 839, in evaluate
    print_node_on_error=self.rec_eval_print_node_on_error)

  File "/home/robin/anaconda3/lib/python3.5/site-packages/hyperopt/pyll/base.py", line 913, in rec_eval
    rval = scope._impls[node.name](*args, **kwargs)

  File "/home/robin/Documents/Python/hyperopt-sklearn/hpsklearn/components.py", line 18, in sklearn_SVC
    return sklearn.svm.SVC(*args, **kwargs)

  File "/home/robin/anaconda3/lib/python3.5/site-packages/sklearn/svm/classes.py", line 543, in __init__
    random_state=random_state)

  File "/home/robin/anaconda3/lib/python3.5/site-packages/sklearn/svm/base.py", line 516, in __init__
    random_state=random_state)

  File "/home/robin/anaconda3/lib/python3.5/site-packages/sklearn/svm/base.py", line 86, in __init__
    raise ValueError(msg)

ValueError: The gamma value of 0.0 is invalid. Use 'auto' to set gamma to a value of 1 / n_features.
``
I understand the error, however, i'm not sure of where to fix this one.

**Comment by bjkomer** [comment]
fixed in #107
